crates/supervisor/core/src/supervisor.rs

```diff
@@ -256,13 +288,68 @@ impl SupervisorService for Supervisor {
         Ok(self.database_factory.get_db(chain)?.get_safety_head_ref(SafetyLevel::Finalized)?)
     }
 
-    async fn check_access_list(
+    fn check_access_list(
         &self,
-        _inbox_entries: Vec<B256>,
-        _min_safety: SafetyLevel,
-        _executing_descriptor: ExecutingDescriptor,
+        inbox_entries: Vec<B256>,
+        min_safety: SafetyLevel,
+        executing_descriptor: ExecutingDescriptor,
     ) -> Result<(), SupervisorError> {
-        Err(SupervisorError::Unimplemented)
+        let access_list = parse_access_list(inbox_entries)?;
+        let message_expiry_window = self.config.dependency_set.get_message_expiry_window();
+        let timeout = executing_descriptor.timeout.unwrap_or(0);
+        let executing_ts_with_duration = executing_descriptor.timestamp.saturating_add(timeout);
+
+        for access in &access_list {
+            // Check all the invariants for each message
+            // Ref: https://github.com/ethereum-optimism/specs/blob/main/specs/interop/derivation.md#invariants
+
+            // TODO: support 32 bytes chain id and convert to u64 via dependency set to be usable
+            // across services
+            let initiating_chain_id =
+                u64::from_be_bytes(access.chain_id[24..32].try_into().unwrap());
+
+            // TODO: Extend the `ExecutingDescriptor` to accept chain_id.
+            // And set executing_chain_id as initiating_chain_id only for backward compat.
+            let executing_chain_id = initiating_chain_id;
```

Comment:
Can we remove this for now since it's not part of the spec? 
We can keep the todo to discuss the same with the team.

Reply:
It was missing from the spec. A PR is already there to fix this
https://github.com/op-rs/kona/pull/2173